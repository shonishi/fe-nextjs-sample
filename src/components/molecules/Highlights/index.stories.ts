import type { Meta, StoryObj } from '@storybook/react';
import Highlights from '.';

const meta: Meta<typeof Highlights> = {
  component: Highlights,
};

export default meta;
type Story = StoryObj<typeof Highlights>;

export const MultipleElement: Story = {
  args: {
    highlights: [
      'お風呂に浮く',
      '握ると音が鳴る',
      'シリコン製で丈夫',
      'お子様の筋力強化にも',
    ],
  },
};

export const SingleElement: Story = {
  args: {
    highlights: ['お風呂に浮く'],
  },
};

export const EmptyElement: Story = {
  args: {
    highlights: [],
  },
};
