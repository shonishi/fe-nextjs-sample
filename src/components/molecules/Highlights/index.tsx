interface Props {
  highlights: string[];
}

export default function Highlights({ highlights }: Props) {
  return (
    <div className="mt-10">
      <h3 className="text-sm font-medium text-gray-900">Highlights</h3>
      <div className="mt-4">
        <ul role="list" className="list-disc space-y-2 pl-4 text-sm">
          {highlights.map((highlight) => (
            <li key={highlight} className="text-gray-400">
              <span className="text-gray-600">{highlight}</span>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}
