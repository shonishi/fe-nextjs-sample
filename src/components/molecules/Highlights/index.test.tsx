import Highlights from '.';
import renderer from 'react-test-renderer';

describe('Highlights', () => {
  it('highlightsの要素が複数の場合、全て表示されること', async () => {
    const tree = renderer
      .create(
        <Highlights
          {...{
            highlights: [
              'お風呂に浮く',
              '握ると音が鳴る',
              'シリコン製で丈夫',
              'お子様の筋力強化にも',
            ],
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('highlightsの要素が1つの場合、1つだけ表示されること', async () => {
    const tree = renderer
      .create(<Highlights {...{ highlights: ['お風呂に浮く'] }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('highlightsが空の場合、何も表示されないこと', async () => {
    const tree = renderer
      .create(<Highlights {...{ highlights: [] }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
