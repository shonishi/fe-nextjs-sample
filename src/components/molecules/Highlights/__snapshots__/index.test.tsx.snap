// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlights highlightsが空の場合、何も表示されないこと 1`] = `
<div
  className="mt-10"
>
  <h3
    className="text-sm font-medium text-gray-900"
  >
    Highlights
  </h3>
  <div
    className="mt-4"
  >
    <ul
      className="list-disc space-y-2 pl-4 text-sm"
      role="list"
    />
  </div>
</div>
`;

exports[`Highlights highlightsの要素が1つの場合、1つだけ表示されること 1`] = `
<div
  className="mt-10"
>
  <h3
    className="text-sm font-medium text-gray-900"
  >
    Highlights
  </h3>
  <div
    className="mt-4"
  >
    <ul
      className="list-disc space-y-2 pl-4 text-sm"
      role="list"
    >
      <li
        className="text-gray-400"
      >
        <span
          className="text-gray-600"
        >
          お風呂に浮く
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Highlights highlightsの要素が複数の場合、全て表示されること 1`] = `
<div
  className="mt-10"
>
  <h3
    className="text-sm font-medium text-gray-900"
  >
    Highlights
  </h3>
  <div
    className="mt-4"
  >
    <ul
      className="list-disc space-y-2 pl-4 text-sm"
      role="list"
    >
      <li
        className="text-gray-400"
      >
        <span
          className="text-gray-600"
        >
          お風呂に浮く
        </span>
      </li>
      <li
        className="text-gray-400"
      >
        <span
          className="text-gray-600"
        >
          握ると音が鳴る
        </span>
      </li>
      <li
        className="text-gray-400"
      >
        <span
          className="text-gray-600"
        >
          シリコン製で丈夫
        </span>
      </li>
      <li
        className="text-gray-400"
      >
        <span
          className="text-gray-600"
        >
          お子様の筋力強化にも
        </span>
      </li>
    </ul>
  </div>
</div>
`;
